packages/superset-ui-core/src/query/types/AnnotationLayer.ts

```diff
@@ -54,6 +56,11 @@ export type IntervalAnnotationLayer = BaseAnnotationLayer &
     annotationType: 'INTERVAL';
   };
 
+export type TableAnnotationLayer = BaseAnnotationLayer &
+  TableSourceAnnotationLayer & {
+    annotationType: 'EVENT' | 'INTERVAL';
```

Comment:
Can we give `annotationType` its own type? Either a string union literal type or `Enum` would be fine.

```ts
type AnnotationType = 'Event' | 'Interval' | ...

interface BaseAnnotationLayer {
  annotationType: AnnotationType
}
```

Reply:
Good idea